pubsub/envelope.go

```diff
@@ -103,6 +103,27 @@ func FromCloudEvent(cloudEvent []byte, topic, pubsub, traceID string) (map[strin
 	m[TopicField] = topic
 	m[PubsubField] = pubsub
 
+	// specify default value if it's unspecified from the original CloudEvent
+	if m[IDField] == nil {
```

Comment:
dapr/dapr#3356 fixes the issue in the runtime. I would still have this change, but I think we should remove the `""` defaults for ID and Type and keep the rest. IMO `nil`/not set is better than an empty string when the field has really has no value.

Reply:
@pkedy pls. take a look again.